Gems/Multiplayer/Code/Source/NetworkEntity/NetworkEntityManager.cpp

```diff
@@ -467,8 +472,87 @@ namespace Multiplayer
         return netEntityId;
     }
 
-    void NetworkEntityManager::OnRootSpawnableAssigned(
-        [[maybe_unused]] AZ::Data::Asset<AzFramework::Spawnable> rootSpawnable, [[maybe_unused]] uint32_t generation)
+    AZStd::unique_ptr<AzFramework::EntitySpawnTicket> NetworkEntityManager::RequestNetSpawnableInstantiation(
+        const AZ::Data::Asset<AzFramework::Spawnable>& rootSpawnable, const AZStd::vector<AZ::Entity*>& entities)
+    {
+        if (entities.empty())
+        {
+            AZ_Error("NetworkEntityManager", false,
+                "RequestNetSpawnableInstantiation: No entities in the spawnable %s", rootSpawnable.GetHint().c_str());
+            return nullptr;
+        }
+
+        // The first entity in every spawnable is the root one 
+        const AZ::Entity* rootEntity = *entities.begin();
+        if(!rootEntity)
+        {
+            AZ_Error("NetworkEntityManager", false,
+                "RequestNetSpawnableInstantiation: Root entity is null in the spawnable %s", rootSpawnable.GetHint().c_str());
+            return nullptr;
+        }
+
+        const auto* holderComponent = rootEntity->FindComponent<NetworkSpawnableHolderComponent>();
+        if(!holderComponent)
+        {
+            // This spawnable doesn't have a corresponding network spawnable.
+            return nullptr;
+        }
+
+        AzFramework::TransformComponent* rootEntityTransform =
+            rootEntity->FindComponent<AzFramework::TransformComponent>();
+        if (!rootEntityTransform)
+        {
+            AZ_Error("NetworkEntityManager", false,
+                "RequestNetSpawnableInstantiation: Root entity has no transform in the spawnable %s", rootSpawnable.GetHint().c_str());
+            return nullptr;
+        }
+
+        // Retrieve the corresponding network spawnable asset
+        AZ::Data::Asset<AzFramework::Spawnable> netSpawnableAsset = holderComponent->GetNetworkSpawnableAsset();
+
+        // Prepare the parameters for the spawning process
+        AzFramework::SpawnAllEntitiesOptionalArgs optionalArgs;
+        optionalArgs.m_priority = AzFramework::SpawnablePriority_High;
+
+        const AZ::Name netSpawnableName =
+            AZ::Interface<INetworkSpawnableLibrary>::Get()->GetSpawnableNameFromAssetId(netSpawnableAsset.GetId());
+
+        // Pre-insertion callback allows us to do network-specific setup for the entities before they are added to the scene
+        optionalArgs.m_preInsertionCallback = [netSpawnableName, rootTransform = rootEntityTransform->GetWorldTM()]
+            (AzFramework::EntitySpawnTicket::Id, AzFramework::SpawnableEntityContainerView entities)
+        {
+            bool shouldUpdateTransform = (rootTransform.IsClose(AZ::Transform::Identity()) == false);
+
+            for (uint32_t netEntityIndex = 0, entitiesSize = aznumeric_cast<uint32_t>(entities.size());
```

Comment:
nit: While the compile may optimize, shouldn't we always prefix increment?

Reply:
There's no difference in the resulting compiled code in this case since it's built-in type, but I don't mind changing if the prefix version looks more pleasant for the people reading the code.